server/src/chunkworker.rs

```diff
@@ -145,21 +155,42 @@ fn load_chunk(worker: &mut ChunkWorker, pos: ChunkPosition) -> Reply {
     let file = worker.open_regions.get_mut(&rpos).unwrap();
     let handle = &mut file.handle;
 
-    load_chunk_from_handle(pos, handle)
+    load_chunk_from_handle(
+        pos,
+        handle,
+        &Arc::from(worker.sender.clone()),
+        &worker.world_generator,
+    )
 }
 
-fn load_chunk_from_handle(pos: ChunkPosition, handle: &mut RegionHandle) -> Reply {
+fn load_chunk_from_handle(
+    pos: ChunkPosition,
+    handle: &mut RegionHandle,
+    sender: &Arc<Sender<Reply>>,
+    generator: &Arc<dyn WorldGenerator>,
+) -> Option<Reply> {
     let result = handle.load_chunk(pos);
 
     match result {
-        Ok(chunk) => (pos, Ok(chunk)),
+        Ok(chunk) => Some((pos, Ok(chunk))),
         Err(e) => match e {
-            region::Error::ChunkNotExist => (pos, Err(Error::ChunkNotExist)),
-            err => (pos, Err(Error::LoadError(err))),
+            region::Error::ChunkNotExist => {
+                let sender = Arc::clone(sender);
```

Comment:
Rather than wrapping the channel in an `Arc` and cloning it, you can just clone the channel directly. https://docs.rs/crossbeam/0.7.2/crossbeam/channel/index.html#sharing-channels